logstash-core/lib/logstash/agent.rb

```diff
@@ -260,16 +264,17 @@ def create_pipeline(settings, config=nil)
     end
   end
 
-  def fetch_config(settings)
-    @config_loader.format_config(settings.get("path.config"), settings.get("config.string"))
+  def fetch_config
+    # TODO(ph) multiple pipeline, we only support one config for now
+    @source_loader.fetch.first
```

Comment:
Just realized, we really need that because lets say we delete the remote config.. Logstash need to stop the corresponding pipeline and wait.